src/coreclr/jit/emitriscv64.cpp

```diff
@@ -2182,672 +2725,462 @@ ssize_t emitter::emitOutputInstrJumpDistance(const BYTE* dst, const BYTE* src, c
     return distVal;
 }
 
-/*****************************************************************************
- *
- *  Append the machine code corresponding to the given instruction descriptor
- *  to the code block at '*dp'; the base of the code block is 'bp', and 'ig'
- *  is the instruction group that contains the instruction. Updates '*dp' to
- *  point past the generated code, and returns the size of the instruction
- *  descriptor in bytes.
- */
+static constexpr size_t NBitMask(uint8_t bits)
+{
+    return (static_cast<size_t>(1) << bits) - 1;
+}
 
-size_t emitter::emitOutputInstr(insGroup* ig, instrDesc* id, BYTE** dp)
+template <uint8_t MaskSize>
+static ssize_t LowerNBitsOfWord(ssize_t word)
 {
-    BYTE* const       dst    = *dp;
-    BYTE*             dstRW  = *dp + writeableOffset;
-    BYTE*             dstRW2 = dstRW + 4; // addr for updating gc info if needed.
-    const BYTE* const odstRW = dstRW;
-    const BYTE* const odst   = *dp;
-    code_t            code   = 0;
-    instruction       ins;
-    size_t            sz; // = emitSizeOfInsDsc(id);
+    static_assert(MaskSize < 32, "Given mask size is bigger than the word itself");
+    static_assert(MaskSize > 0, "Given mask size cannot be zero");
 
-    assert(REG_NA == (int)REG_NA);
+    static constexpr size_t kMask = NBitMask(MaskSize);
 
-    insOpts insOp = id->idInsOpt();
+    return word & kMask;
+}
```

Comment:
Isn't it strange that `MaskSize` can only be up to 32 when the word is a 64 bit value?